cmake/modules/engine_config.cmake

```diff
@@ -0,0 +1,30 @@
+option(BUILD_LIBSCAP_MODERN_BPF "Enable modern bpf probe" OFF)
+option(CREATE_TEST_TARGETS "Enable make-targets for unit testing" ON)
+
+if(CREATE_TEST_TARGETS)
+	# Add engine only used for testing
+	set(HAS_ENGINE_TEST_INPUT On)
+endif()
+
+set(HAS_ENGINE_NODRIVER On)
+set(HAS_ENGINE_SAVEFILE On)
+set(HAS_ENGINE_SOURCE_PLUGIN On)
+
+if(CMAKE_SYSTEM_NAME MATCHES "Linux")
+	set(HAS_ENGINE_KMOD On)
+	set(HAS_ENGINE_UDIG On)
+	set(HAS_ENGINE_BPF On)
+
+	if(BUILD_LIBSCAP_MODERN_BPF)
```

Comment:
I would move the `option(BUILD_LIBSCAP_MODERN_BPF "Enable modern bpf probe" OFF)` inside the Linux check.